compiler/parser/parser.peg

```diff
@@ -654,13 +654,18 @@ DerefExpr
   = first:DotID rest:(Deref)* {
       RETURN(makeBinaryExprChain(first, rest))
     }
+  / first:This rest:(Deref)* {
+      RETURN(makeBinaryExprChain(first, rest))
+    }
   / first:Identifier rest:(Deref)* {
       RETURN(makeBinaryExprChain(first, rest))
     }
   / "." {
       RETURN(MAP("kind":"Root"))
     }
 
+This = "this" { RETURN(MAP("kind":"Root")) }
```

Comment:
Should this be called `ThisToken`?

Reply:
If it's called a token, then the action should be to return TEXT but that's not the case. If this is confusing, we could say  `RootRecord = "this"` instead.